tests/testSpanSets.cc

```diff
@@ -0,0 +1,397 @@
+}
+
+void testComputeShape() {
+    auto SpanSetShape = geom::SpanSet::spanSetFromShape(1);
+    auto quad = SpanSetShape.computeShape();
+
+    BOOST_CHECK(quad.getIxx() == 0.4);
+    BOOST_CHECK(quad.getIyy() == 0.4);
+    BOOST_CHECK(quad.getIxy() == 0);
+}
+
+void testDilate() {
+    // Only need to test use of a stencil as it delegates to the other function overload
+    auto SpanSetPreDilate = geom::SpanSet::spanSetFromShape(1);
+    // This should dilate the shape by 1
+    auto SpanSetPostDilate = SpanSetPreDilate.dilate(1);
+
+    BOOST_CHECK(SpanSetPostDilate.getBBox().getMinX() == -2);
+    BOOST_CHECK(SpanSetPostDilate.getBBox().getMinY() == -2);
+}
+
+void testErode() {
+    // Only need to test use of a stencil as it delegates to the other function overload
+    auto SpanSetPreErode = geom::SpanSet::spanSetFromShape(2);
+    // This should erode the shape by 1
+    auto SpanSetPostErode = SpanSetPreErode.erode(1);
+
+    BOOST_CHECK(SpanSetPostErode.getBBox().getMinX() == -1);
+    BOOST_CHECK(SpanSetPostErode.getBBox().getMinY() == -1);
+}
+
+void testFlatten() {
+    // Test version without output array, as this simply delegates
+    // Create an array and initialize it to 9
+    ndarray::Array<int, 2, 2> input = ndarray::allocate(ndarray::makeVector(6, 6));
+    input.deep() = 9;
+
+    // overload a few pixels with the value 1
+    input[1][1] = 1;
+    input[1][2] = 1;
+    input[2][1] = 1;
+    input[2][2] = 1;
+    // Create a spanSet for those points
+    std::vector<geom::Span> spanVector = {geom::Span(0, 0, 1), geom::Span(1, 0, 1)};
+    geom::SpanSet SpnSt(spanVector);
+    auto flatArr = SpnSt.flatten(input, geom::Point2I(-1, -1));
+    for (auto const &  val : flatArr) {
+        BOOST_CHECK(val == 1);
+    }
+}
+
+void testUnflatten() {
+    // Test version without output array, as this simply delegates
+    ndarray::Array<int, 1, 1> input = ndarray::allocate(6);
+    input.deep() = 9;
+
+    // Create a SpanSet which has six points int it
+    std::vector<geom::Span> spanVector = {geom::Span(9, 2, 3), geom::Span(10, 3, 4), geom::Span(11, 2, 3)};
+    geom::SpanSet SpnSt(spanVector);
+    auto output = SpnSt.unflatten(input);
+
+    // Loop over each point in the array correspointing to the spanset, minux xy0 (the min of the SpanSet)
+    // verify the array contains the value 9
+    for (auto const & spn : SpnSt) {
+        for (auto const & pt : spn) {
+            BOOST_CHECK(output[pt.getY()-9][pt.getX()-2] == 9);
+        }
+    }
+    // Verify that a point outside the vector has the value zero
+    BOOST_CHECK(output[0][2] == 0);
+}
+
+std::pair<lsst::afw::image::Mask<lsst::afw::image::MaskPixel>, geom::SpanSet> populateMask() {
+    // Create a mask and populate it with the value 2
+    lsst::afw::image::Mask<lsst::afw::image::MaskPixel> msk(10, 10);
+    auto spnSt = geom::SpanSet::spanSetFromShape(3).shiftedBy(5, 5);
+    spnSt.setMask(msk, static_cast<lsst::afw::image::MaskPixel>(2));
```

Comment:
The reason this was in this function is simply to share code between multiple tests. This does not cause any loss of functionality or clarity, and so I would push back on the necessity of moving it